app/code/Magento/InventoryConfiguration/Model/StockConfigurationValidator.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: dev
+ * Date: 08.02.18
+ * Time: 23:54
+ */
+
+namespace Magento\InventoryConfiguration\Model;
+
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Catalog\Model\ResourceModel\Product as ProductResourceModel;
+use Magento\CatalogInventory\Model\Configuration;
+use Magento\CatalogInventory\Model\Stock\StockItemRepository as LegacyStockItemRepository;
+use Magento\CatalogInventory\Model\Stock\Item as LegacyStockItem;
+use Magento\CatalogInventory\Api\Data\StockItemInterface;
+use Magento\CatalogInventory\Api\StockItemCriteriaInterfaceFactory;
+
+class StockConfigurationValidator implements StockConfigurationInterface
+{
+    /**
+     * @var StockConfigurationInterface[]
+     */
+    protected $validators;
+
+    /**
+     * @var StockItemCriteriaInterfaceFactory
+     */
+    private $stockItemCriteriaFactory;
+
+    /**
+     * @var LegacyStockItemRepository
+     */
+    protected $legacyStockItemRepository;
+
+    /**
+     * @var Configuration
+     */
+    protected $configuration;
+
+    /**
+     * @var ProductResourceModel
+     */
+    protected $productResource;
+
+    /**
+     * StockValidationComposer constructor.
+     * @param StockConfigurationInterface[] $validators
+     * @param Configuration $configuration
+     * @param LegacyStockItemRepository $legacyStockItemRepository
+     * @param StockItemCriteriaInterfaceFactory $stockItemCriteriaFactory
+     * @param ProductResourceModel $productResource
+     * @throws LocalizedException
+     */
+    public function __construct(
+        array $validators=[],
+        Configuration $configuration,
+        LegacyStockItemRepository $legacyStockItemRepository,
+        StockItemCriteriaInterfaceFactory $stockItemCriteriaFactory,
+        ProductResourceModel $productResource
+    ) {
+        $this->configuration = $configuration;
+        $this->legacyStockItemRepository = $legacyStockItemRepository;
+        $this->productResource = $productResource;
+        $this->stockItemCriteriaFactory = $stockItemCriteriaFactory;
+
+        foreach ($validators as $validator) {
+            if (!$validator instanceof StockConfigurationInterface) {
+                throw new LocalizedException(
+                    __('Validator must implement StockConfigurationInterface.')
+                );
+            }
+        }
+        $this->validators = $validators;
```

Comment:
you don't use these 
`$this->validators` 
at all, so even if 3PD will add some more validators using DI preferences - they would not be used

Reply:
I just left old implementation here. I need to discuss with you, how this will work.